src/js/components/DataTable/Body.js

```diff
@@ -157,7 +163,7 @@ const Row = memo(
         <StyledDataTableRow key={`${index.toString()}_expand`}>
           {(selected || onSelect) && <TableCell />}
           <TableCell colSpan={columns.length + 1}>
-            {rowDetails(data[index])}
+            {rowDetails.render ? rowDetails.render(datum) : rowDetails(datum)}
```

Comment:
are we sure that datum would be equivalent to `data[index]` or why did you change from `data[index]`

Reply:
Should be. Everything above this including ExpanderCell, selection Cell, column Cells, etc pass the `datum`. `datum` basically gets set by line 294 of InfiniteScroll.js:
```jsx
items.slice(firstIndex, lastIndex + 1).forEach((item, index) => {
    const itemsIndex = firstIndex + index;
    const child = children(item, itemsIndex);
    result.push(child);
  });
```
the `item` passed to `children` as the first parameter is basically `items[index]` with index corrected for the pagination slice.

It seemed simpler to rely on `datum` consistently in each of these spots.